src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -7,25 +7,31 @@ using LinearAlgebra.BLAS: gemv, gemv!, gemm!, trsm!, axpy!, ger!
 
 function rrule(::typeof(svd), X::AbstractMatrix{<:Real})
     F = svd(X)
-    ∂X = Rule() do Ȳ::NamedTuple{(:U,:S,:V)}
-        svd_rev(F, Ȳ.U, Ȳ.S, Ȳ.V)
+    function svd_pullback(Ȳ::NamedTuple{(:U,:S,:V)})
+        ∂X = @thunk(svd_rev(F, Ȳ.U, Ȳ.S, Ȳ.V))
+        return (NO_FIELDS, ∂X)
     end
-    return F, ∂X
+    return F, svd_pullback
 end
 
 function rrule(::typeof(getproperty), F::SVD, x::Symbol)
-    if x === :U
-        rule = Ȳ->(U=Ȳ, S=zero(F.S), V=zero(F.V))
-    elseif x === :S
-        rule = Ȳ->(U=zero(F.U), S=Ȳ, V=zero(F.V))
-    elseif x === :V
-        rule = Ȳ->(U=zero(F.U), S=zero(F.S), V=Ȳ)
-    elseif x === :Vt
-        # TODO: This could be made to work, but it'd be a pain
-        throw(ArgumentError("Vt is unsupported; use V and transpose the result"))
+    function getproperty_svd_pullback(Ȳ)
+        if x === :U
+            ∂ = @thunk((; U=Ȳ, S=(zero(F.S)), V=(zero(F.V))))
+        elseif x === :S
+            ∂ = @thunk((; U=(zero(F.U)), S=Ȳ, V=(zero(F.V))))
+        elseif x === :V
+            ∂ = @thunk((; U=(zero(F.U)), S=(zero(F.S)), V=Ȳ))
+        elseif x === :Vt
+            # TODO: This could be made to work, but it'd be a pain
```

Comment:
We should have an issue to make this work

Reply:
Feel free to open one. (not changed in this PR)

Reply:
https://github.com/JuliaDiff/ChainRules.jl/issues/106